src/libraries/System.Memory/tests/Span/Fill.cs

```diff
@@ -147,5 +150,53 @@ public static unsafe void FillNativeBytes()
                 Marshal.FreeHGlobal(new IntPtr(ptr));
             }
         }
+
+        public static IEnumerable<object[]> FillData
+        {
+            get
+            {
+                yield return new object[] { typeof(sbyte), (sbyte)0x20 };
+                yield return new object[] { typeof(byte), (byte)0x20 };
+                yield return new object[] { typeof(bool), true };
+                yield return new object[] { typeof(short), (short)0x1234 };
+                yield return new object[] { typeof(ushort), (ushort)0x1234 };
+                yield return new object[] { typeof(char), 'x' };
+                yield return new object[] { typeof(int), (int)0x12345678 };
+                yield return new object[] { typeof(uint), (uint)0x12345678 };
+                yield return new object[] { typeof(long), (long)0x0123456789abcdef };
+                yield return new object[] { typeof(ulong), (ulong)0x0123456789abcdef };
+                yield return new object[] { typeof(nint), unchecked((nint)0x0123456789abcdef) };
+                yield return new object[] { typeof(nuint), unchecked((nuint)0x0123456789abcdef) };
+                yield return new object[] { typeof(float), 1.0f };
+                yield return new object[] { typeof(double), 1.0d };
+                yield return new object[] { typeof(string), "Hello world!" }; // shouldn't go down SIMD path
+                yield return new object[] { typeof(decimal), 1.0m }; // shouldn't go down SIMD path
```

Comment:
Why not? `decimal` is a blittable value type that is 128-bits in length, it (like `Guid`) seems like a perfect candidate (outside the possibility of it being "rare")